src/main.rs

```diff
@@ -25,7 +28,28 @@ async fn main() -> Result<(), Box<dyn Error>> {
     let host = IpAddr::from_str(&host_raw)?;
     let port = env::var("PORT").unwrap_or_else(|_| "80".into()).parse()?;
 
-    let queue = LocalQueue::new();
+    let queue: Arc<Box<dyn Queue>> = match env::var("TOKEN") {
+        Ok(token) => {
+            let http_client = Client::new(token);
+            let gateway = http_client
+                .gateway()
+                .authed()
+                .await
+                .expect("Cannot fetch gateway information");
+            Arc::new(Box::new(
+                LargeBotQueue::new(
+                    gateway
+                        .session_start_limit
+                        .max_concurrency
+                        .try_into()
+                        .unwrap(),
+                    &http_client,
+                )
+                .await,
+            ))
+        }
+        Err(_) => Arc::new(Box::new(LocalQueue::new())),
+    };
```

Comment:
Maybe this could be done with a `if let Ok(token) = env::var("DISCORD_TOKEN") { ... }` instead to make it nicer

Reply:
Done.